src/lib.rs

```diff
@@ -523,6 +524,22 @@ impl<'a, R: Read> Iterator for FilesMut<'a, R> {
     }
 }
 
+impl Clone for Header {
+    fn clone(&self) -> Header {
+        let mut header = Header::new();
+        unsafe { ptr::copy_nonoverlapping(self, &mut header as *mut Header, 1) };
+        header
+    }
+}
+
+impl PartialEq for Header {
+    fn eq(&self, other: &Header) -> bool {
+        self.as_bytes()[..] == other.as_bytes()[..]
```

Comment:
Actually, I'll remove Eq/PartialEq and come back to it when you can see my thoughts on a higher level api and extended headers etc.